streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -737,12 +737,15 @@ private void checkIfUnexpectedUserSpecifiedConsumerConfig(final Map<String, Obje
 
         // add configs required for stream partition assignor
         consumerProps.put(REPLICATION_FACTOR_CONFIG, getInt(REPLICATION_FACTOR_CONFIG));
+        consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
         consumerProps.put(NUM_STANDBY_REPLICAS_CONFIG, getInt(NUM_STANDBY_REPLICAS_CONFIG));
         consumerProps.put(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, StreamPartitionAssignor.class.getName());
         consumerProps.put(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG, getLong(WINDOW_STORE_CHANGE_LOG_ADDITIONAL_RETENTION_MS_CONFIG));
-        consumerProps.put(APPLICATION_SERVER_CONFIG, getString(APPLICATION_SERVER_CONFIG));
-        final AdminClientConfig config = new AdminClientConfig(getClientPropsWithPrefix(ADMIN_CLIENT_PREFIX, AdminClientConfig.configNames()));
-        consumerProps.put(adminClientPrefix(AdminClientConfig.RETRIES_CONFIG), config.getInt(AdminClientConfig.RETRIES_CONFIG));
+
+        // add admin and topic configs required for creating topics
+        consumerProps.put(adminClientPrefix(RETRIES_CONFIG), getInt(RETRIES_CONFIG));
+        consumerProps.putAll(originalsWithPrefix(TOPIC_PREFIX, false));
+        consumerProps.putAll(originalsWithPrefix(ADMIN_CLIENT_PREFIX, false));
```

Comment:
Do we need to pass in all admin client configs? Actually, we only need retries.

Also, I am not sure if we would set the correct default here. If nothing is specified, we would use `StreamsConfig` retry as default, but we actually should set `AdminClientConfig` default -- atm, both might be the same so it doesn't matter too much. Just is doesn't seems to be "correct" (in a very strong sense).

Reply:
That's a good point: I will revert the changes.